This patch ensures the target triple that is passed during the compiler build is
respected, instead of mangling it.

Introduced here: http://svnweb.freebsd.org/changeset/base/221503

Index: lib/Support/Unix/Host.inc
===================================================================
--- lib/Support/Unix/Host.inc
+++ lib/Support/Unix/Host.inc
@@ -26,6 +26,11 @@
 
 using namespace llvm;
 
+#ifdef __FreeBSD__
+std::string sys::getDefaultTargetTriple() {
+  return LLVM_DEFAULT_TARGET_TRIPLE;
+}
+#else // __FreeBSD__
 static std::string getOSVersion() {
   struct utsname info;
 
@@ -61,3 +66,4 @@ std::string sys::getDefaultTargetTriple() {
 
   return Triple;
 }
+#endif // __FreeBSD__
